bfvmm/include/hve/arch/intel_x64/ept/mmap.h

```diff
@@ -679,6 +825,55 @@ class mmap
 
 private:
 
+    std::pair<std::reference_wrapper<entry_type>, uintptr_t>
+    entry(void *virt_addr, bool should_lock)
+    {
+        if (should_lock) {
+            m_mutex.lock();
+        }
+
```

Comment:
need to add a gsl::finally call to make sure its unlocked if an exception if thrown before the unlock() call below:
```cpp
if (should_lock) {
    m_mutex.lock();
}

auto unlock = gsl::finally([&]() { if (should_lock) m_mutex.unlock(); });
```

Now when unlock goes out of scope, either through normal return or via exception, the mutex is unlocked. With this you can remove the unlock() call below

Reply:
Thank you for the feedback. I resolved this issue in https://github.com/Bareflank/hypervisor/pull/922/commits/37f0095d263f6cd950223f34f0b46ee55208682d